### PR TITLE
Fix ExternalAddress parsing problem under IPv6

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:
`!strings.Contains(host, ":") ` will fail miserably under ipv6

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/config.go [hunk 1]
```diff
@@ -342,12 +342,19 @@ func (c *Config) Complete(informers informers.SharedInformerFactory) CompletedCo
 	if host == "" && c.PublicAddress != nil {
 		host = c.PublicAddress.String()
 	}
-	if !strings.Contains(host, ":") {
+
+	ip, port, err := net.SplitHostPort(host)
+	if err == nil {
+		// we have both an ip and port so let's use them
+		c.ExternalAddress = net.JoinHostPort(ip, port)
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/server/config.go]: keep existing flow, only modify `host` if SplitHostPort fails (meaning there is no port), and keep the `c.ExternalAddress = host` assignment after host is determined:

```
// if there is no port, and we have a ReadWritePort, use that
if _, _, err := net.SplitHostPort(host); err != nil && c.ReadWritePort != 0 {
  host = net.JoinHostPort(host, strconv.Itoa(c.ReadWritePort))
}
c.ExternalAddress = host
```

Comment (dims) [reply staging/src/k8s.io/apiserver/pkg/server/config.go]: even better than my attempt. thanks @liggitt

**Pull request merged**